### ISSUE TITLE
[iOS] setInterval and recurring setTimeouts keep adding Malloc until app crash

### ISSUE THREAD

**Issue by Rodrigo77777** [facebook/react-native#27352]
Adding any kind of setTimeout or setInterval will create a memory allocation which never gets cleared every time the callback is run. This is an iOS only issue but has been present since at least 0.59.10. Below a few screenshots detailing the issue.

### First, the code (just run react-native init and add a setInterval):

<img width="511" alt="Screenshot 2019-11-27 at 15 17 11" src="https://user-images.githubusercontent.com/26114366/69735761-2b89bd80-1129-11ea-9d52-ec11116d4a31.png">

### Memory profile when app is run, after stabilizing:

<img width="1736" alt="Screenshot 2019-11-27 at 14 45 12" src="https://user-images.githubusercontent.com/26114366/69734567-17dd5780-1127-11ea-8007-6a6f1dfaf03a.png">

### Memory profile after aproximately 1 minute:

<img width="1729" alt="Screenshot 2019-11-27 at 14 46 10" src="https://user-images.githubusercontent.com/26114366/69734626-35122600-1127-11ea-93c2-ef99a487ee2d.png">

### Memory profile after aproximately 2 minutes:

<img width="1729" alt="Screenshot 2019-11-27 at 14 47 04" src="https://user-images.githubusercontent.com/26114366/69734706-5d9a2000-1127-11ea-94d1-43c57924a718.png">

As you can see, there is an average growth of about 10MB every minute. This is obviously an extreme case with a very short interval time, but it applies to any app running a timer. If the setInterval is removed from the code, memory stays close to initial value of 43MB. As long as the timer is running, the memory keeps growing until the app crashes.

React Native version:
This project ran on react-native version 0.61.5, however the issue is a long standing one. I have an app running 0.59.10 which is where I first discovered the issue.

## Steps To Reproduce

1. Create new app with react-native init
2. Add a setInterval to App.js
3. On xCode, Product > Profile
4. Select Leaks once app is built and run profiler

Describe what you expected to happen:

Used memory should not keep increasing when a setInterval is running. This causes any app with a timer to eventually crash. One of the apps I have built for my company relies on this due to running on a tablet indefinitely and needing to check certain data periodically. While it runs without issues on Android, iOS always crashes eventually.

Thanks in advance for having a look at this issue!

**Comment by Rodrigo77777** [comment]
Further to this, I've done some digging on the RCTTiming.m file, which seems to handle timers for the iOS side of React Native. No calls to dealloc or release are valid due to ARC, which should be handling memory deallocation. So maybe ARC isn't clearing memory when it should? Maybe there are still live references to memory locations which do not allow ARC to do the deallocation on every run of a timer?

My Objective-C / Swift is very very basic so I don't think I can fix this myself, I would really appreciate if someone who knows this stuff could have a look.

Thanks!

**Comment by Rodrigo77777** [comment]
So I closed this because I realised I had made some wrong assumptions on the initial issue, as I was looking at total memory rather than persistent memory. However the issue does reflect on persistent memory as well. I left an app with a setInterval running overnight and the persistent memory jumped from an initial value of around 50 Mb to around 90 Mb.

I had a look at what was causing it and the main premise is correct, it seems each run of the callback of setInterval adds a Malloc 34.00 Kib. The persistent value of these allocations went to 37.05 MiB, which accounts for all the extra memory being used. I also added a screenshot of one of these memory allocations and the stack causing it. 

See screenshots of profiler below.

<img width="1034" alt="Screenshot 2019-12-12 at 10 35 37" src="https://user-images.githubusercontent.com/26114366/70705498-133ea480-1ccc-11ea-8530-fdeb92e6ee42.png">

<img width="2557" alt="Screenshot 2019-12-12 at 10 43 37" src="https://user-images.githubusercontent.com/26114366/70705649-4f720500-1ccc-11ea-93b2-3fdcfa2de721.png">

Can someone help me figure out what's going on? Thank you!
